// ngx-imagesloaded/bundles/ngx-imagesloaded.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('imagesloaded')) :
    typeof define === 'function' && define.amd ? define('ngx-imagesloaded', ['exports', '@angular/core', 'imagesloaded'], factory) :
    (factory((global['ngx-imagesloaded'] = {}),global.ng.core,null));
}(this, (function (exports,i0,imagesLoaded) { 'use strict';

    imagesLoaded = imagesLoaded && imagesLoaded.hasOwnProperty('default') ? imagesLoaded['default'] : imagesLoaded;

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxImagesloadedService = (function () {
        function NgxImagesloadedService() {
        }
        NgxImagesloadedService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        /** @nocollapse */
        NgxImagesloadedService.ctorParameters = function () { return []; };
        /** @nocollapse */ NgxImagesloadedService.ngInjectableDef = i0.defineInjectable({ factory: function NgxImagesloadedService_Factory() { return new NgxImagesloadedService(); }, token: NgxImagesloadedService, providedIn: "root" });
        return NgxImagesloadedService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxImagesloadedDirective = (function () {
        //
        function NgxImagesloadedDirective(el) {
            this.el = el;
            //
            this.always = new i0.EventEmitter();
            //
            this.done = new i0.EventEmitter();
            //
            this.fail = new i0.EventEmitter();
            //
            this.progress = new i0.EventEmitter();
        }
        //
        /**
         * @return {?}
         */
        NgxImagesloadedDirective.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                var _this = this;
                this.instance = imagesLoaded(this.el.nativeElement, this.options, function (instance) {
                    _this.always.emit(instance);
                });
                this.instance.on('always', function (instance) { return _this.always.emit(instance); });
                this.instance.on('done', function (instance) { return _this.always.emit(instance); });
                this.instance.on('fail', function (instance) { return _this.always.emit(instance); });
                this.instance.on('progress', function (instance, image) {
                    return _this.always.emit({ instance: instance, image: image });
                });
            };
        //
        /**
         * @return {?}
         */
        NgxImagesloadedDirective.prototype.ngOnDestroy = /**
         * @return {?}
         */
            function () {
                var _this = this;
                this.instance.off('always', function (instance) { return _this.always.emit(instance); });
                this.instance.off('doffe', function (instance) { return _this.always.emit(instance); });
                this.instance.off('fail', function (instance) { return _this.always.emit(instance); });
                this.instance.off('progress', function (instance, image) {
                    return _this.always.emit({ instance: instance, image: image });
                });
            };
        NgxImagesloadedDirective.decorators = [
            { type: i0.Directive, args: [{
                        selector: '[imagesLoaded]'
                    },] },
        ];
        /** @nocollapse */
        NgxImagesloadedDirective.ctorParameters = function () {
            return [
                { type: i0.ElementRef }
            ];
        };
        NgxImagesloadedDirective.propDecorators = {
            always: [{ type: i0.Output }],
            done: [{ type: i0.Output }],
            fail: [{ type: i0.Output }],
            progress: [{ type: i0.Output }],
            options: [{ type: i0.Input }]
        };
        return NgxImagesloadedDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxImagesloadedModule = (function () {
        function NgxImagesloadedModule() {
        }
        NgxImagesloadedModule.decorators = [
            { type: i0.NgModule, args: [{
                        declarations: [NgxImagesloadedDirective],
                        exports: [NgxImagesloadedDirective],
                        providers: [NgxImagesloadedService]
                    },] },
        ];
        return NgxImagesloadedModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.NgxImagesloadedService = NgxImagesloadedService;
    exports.NgxImagesloadedModule = NgxImagesloadedModule;
    exports.ɵa = NgxImagesloadedDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
