// ngx-imagesloaded/fesm2015/ngx-imagesloaded.js
import { Injectable, Directive, ElementRef, EventEmitter, Input, Output, NgModule, defineInjectable } from '@angular/core';
import imagesLoaded from 'imagesloaded';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxImagesloadedService {
    constructor() { }
}
NgxImagesloadedService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
NgxImagesloadedService.ctorParameters = () => [];
/** @nocollapse */ NgxImagesloadedService.ngInjectableDef = defineInjectable({ factory: function NgxImagesloadedService_Factory() { return new NgxImagesloadedService(); }, token: NgxImagesloadedService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxImagesloadedDirective {
    /**
     * @param {?} el
     */
    constructor(el) {
        this.el = el;
        //
        this.always = new EventEmitter();
        //
        this.done = new EventEmitter();
        //
        this.fail = new EventEmitter();
        //
        this.progress = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.instance = imagesLoaded(this.el.nativeElement, this.options, instance => {
            this.always.emit(instance);
        });
        this.instance.on('always', instance => this.always.emit(instance));
        this.instance.on('done', instance => this.always.emit(instance));
        this.instance.on('fail', instance => this.always.emit(instance));
        this.instance.on('progress', (instance, image) => this.always.emit({ instance, image }));
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.instance.off('always', instance => this.always.emit(instance));
        this.instance.off('doffe', instance => this.always.emit(instance));
        this.instance.off('fail', instance => this.always.emit(instance));
        this.instance.off('progress', (instance, image) => this.always.emit({ instance, image }));
    }
}
NgxImagesloadedDirective.decorators = [
    { type: Directive, args: [{
                selector: '[imagesLoaded]'
            },] },
];
/** @nocollapse */
NgxImagesloadedDirective.ctorParameters = () => [
    { type: ElementRef }
];
NgxImagesloadedDirective.propDecorators = {
    always: [{ type: Output }],
    done: [{ type: Output }],
    fail: [{ type: Output }],
    progress: [{ type: Output }],
    options: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxImagesloadedModule {
}
NgxImagesloadedModule.decorators = [
    { type: NgModule, args: [{
                declarations: [NgxImagesloadedDirective],
                exports: [NgxImagesloadedDirective],
                providers: [NgxImagesloadedService]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgxImagesloadedService, NgxImagesloadedModule, NgxImagesloadedDirective as ɵa };
