// ngx-imagesloaded/esm5/lib/ngx-imagesloaded.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import imagesLoaded from 'imagesloaded';
var NgxImagesloadedDirective = /** @class */ (function () {
    //
    function NgxImagesloadedDirective(el) {
        this.el = el;
        //
        this.always = new EventEmitter();
        //
        this.done = new EventEmitter();
        //
        this.fail = new EventEmitter();
        //
        this.progress = new EventEmitter();
    }
    //
    /**
     * @return {?}
     */
    NgxImagesloadedDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.instance = imagesLoaded(this.el.nativeElement, this.options, function (instance) {
            _this.always.emit(instance);
        });
        this.instance.on('always', function (instance) { return _this.always.emit(instance); });
        this.instance.on('done', function (instance) { return _this.always.emit(instance); });
        this.instance.on('fail', function (instance) { return _this.always.emit(instance); });
        this.instance.on('progress', function (instance, image) {
            return _this.always.emit({ instance: instance, image: image });
        });
    };
    //
    /**
     * @return {?}
     */
    NgxImagesloadedDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.instance.off('always', function (instance) { return _this.always.emit(instance); });
        this.instance.off('doffe', function (instance) { return _this.always.emit(instance); });
        this.instance.off('fail', function (instance) { return _this.always.emit(instance); });
        this.instance.off('progress', function (instance, image) {
            return _this.always.emit({ instance: instance, image: image });
        });
    };
    NgxImagesloadedDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[imagesLoaded]'
                },] },
    ];
    /** @nocollapse */
    NgxImagesloadedDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    NgxImagesloadedDirective.propDecorators = {
        always: [{ type: Output }],
        done: [{ type: Output }],
        fail: [{ type: Output }],
        progress: [{ type: Output }],
        options: [{ type: Input }]
    };
    return NgxImagesloadedDirective;
}());
export { NgxImagesloadedDirective };
function NgxImagesloadedDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxImagesloadedDirective.prototype.always;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.done;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.fail;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.progress;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.options;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.instance;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.el;
}
