// ngx-imagesloaded/esm2015/lib/ngx-imagesloaded.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import imagesLoaded from 'imagesloaded';
export class NgxImagesloadedDirective {
    /**
     * @param {?} el
     */
    constructor(el) {
        this.el = el;
        //
        this.always = new EventEmitter();
        //
        this.done = new EventEmitter();
        //
        this.fail = new EventEmitter();
        //
        this.progress = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.instance = imagesLoaded(this.el.nativeElement, this.options, instance => {
            this.always.emit(instance);
        });
        this.instance.on('always', instance => this.always.emit(instance));
        this.instance.on('done', instance => this.always.emit(instance));
        this.instance.on('fail', instance => this.always.emit(instance));
        this.instance.on('progress', (instance, image) => this.always.emit({ instance, image }));
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.instance.off('always', instance => this.always.emit(instance));
        this.instance.off('doffe', instance => this.always.emit(instance));
        this.instance.off('fail', instance => this.always.emit(instance));
        this.instance.off('progress', (instance, image) => this.always.emit({ instance, image }));
    }
}
NgxImagesloadedDirective.decorators = [
    { type: Directive, args: [{
                selector: '[imagesLoaded]'
            },] },
];
/** @nocollapse */
NgxImagesloadedDirective.ctorParameters = () => [
    { type: ElementRef }
];
NgxImagesloadedDirective.propDecorators = {
    always: [{ type: Output }],
    done: [{ type: Output }],
    fail: [{ type: Output }],
    progress: [{ type: Output }],
    options: [{ type: Input }]
};
function NgxImagesloadedDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxImagesloadedDirective.prototype.always;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.done;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.fail;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.progress;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.options;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.instance;
    /** @type {?} */
    NgxImagesloadedDirective.prototype.el;
}
