// ngx-imagesloaded/fesm5/ngx-imagesloaded.js
import { Injectable, Directive, ElementRef, EventEmitter, Input, Output, NgModule, defineInjectable } from '@angular/core';
import imagesLoaded from 'imagesloaded';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxImagesloadedService = /** @class */ (function () {
    function NgxImagesloadedService() {
    }
    NgxImagesloadedService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgxImagesloadedService.ctorParameters = function () { return []; };
    /** @nocollapse */ NgxImagesloadedService.ngInjectableDef = defineInjectable({ factory: function NgxImagesloadedService_Factory() { return new NgxImagesloadedService(); }, token: NgxImagesloadedService, providedIn: "root" });
    return NgxImagesloadedService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxImagesloadedDirective = /** @class */ (function () {
    //
    function NgxImagesloadedDirective(el) {
        this.el = el;
        //
        this.always = new EventEmitter();
        //
        this.done = new EventEmitter();
        //
        this.fail = new EventEmitter();
        //
        this.progress = new EventEmitter();
    }
    //
    /**
     * @return {?}
     */
    NgxImagesloadedDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.instance = imagesLoaded(this.el.nativeElement, this.options, function (instance) {
            _this.always.emit(instance);
        });
        this.instance.on('always', function (instance) { return _this.always.emit(instance); });
        this.instance.on('done', function (instance) { return _this.always.emit(instance); });
        this.instance.on('fail', function (instance) { return _this.always.emit(instance); });
        this.instance.on('progress', function (instance, image) {
            return _this.always.emit({ instance: instance, image: image });
        });
    };
    //
    /**
     * @return {?}
     */
    NgxImagesloadedDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.instance.off('always', function (instance) { return _this.always.emit(instance); });
        this.instance.off('doffe', function (instance) { return _this.always.emit(instance); });
        this.instance.off('fail', function (instance) { return _this.always.emit(instance); });
        this.instance.off('progress', function (instance, image) {
            return _this.always.emit({ instance: instance, image: image });
        });
    };
    NgxImagesloadedDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[imagesLoaded]'
                },] },
    ];
    /** @nocollapse */
    NgxImagesloadedDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    NgxImagesloadedDirective.propDecorators = {
        always: [{ type: Output }],
        done: [{ type: Output }],
        fail: [{ type: Output }],
        progress: [{ type: Output }],
        options: [{ type: Input }]
    };
    return NgxImagesloadedDirective;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxImagesloadedModule = /** @class */ (function () {
    function NgxImagesloadedModule() {
    }
    NgxImagesloadedModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [NgxImagesloadedDirective],
                    exports: [NgxImagesloadedDirective],
                    providers: [NgxImagesloadedService]
                },] },
    ];
    return NgxImagesloadedModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgxImagesloadedService, NgxImagesloadedModule, NgxImagesloadedDirective as ɵa };
